src/rtc.c

```diff
@@ -0,0 +1,518 @@
+/**
+ * @file rtc.c
+ * @brief Real-Time Clock Subsystem
+ * @ingroup rtc
+ */
+
+#include "libdragon.h"
+
+/**
+ * @defgroup rtc Real-Time Clock Subsystem
+ * @ingroup libdragon
+ * @brief Joybus real-time clock interface.
+ *
+ * The Joybus real-time clock is an accessory that was only included on
+ * one official cartridge that was only available in Japan. The Joybus RTC
+ * is accessed through the serial interface (SI) similar to EEPROM.
+ * A real RTC uses a battery to power a clock that runs even when the N64
+ * is powered-off and maintains the date, time, and day of the week.
+ *
+ * To check if the real-time clock is available, call #joybus_rtc_is_present.
+ * If detected, it is a good idea to ensure the RTC control block is in a
+ * known-good state by calling the #joybus_rtc_run function before reading
+ * the current time with #joybus_rtc_read_time for the first time.
+ *
+ * The simplest method to change the time is by calling #joybus_rtc_set,
+ * which is a high-level convenience function that prepares the RTC and
+ * writes the time with the proper delays.
+ *
+ * In general, since only one game ever used Joybus RTC (and that game was
+ * later re-released on the GameCube in English), RTC support by emulators
+ * and flash carts tends to be incomplete, inaccurate, or non-existent.
+ * Many emulators do not actually implement the RTC write command and will
+ * always respond with the host system's current local time. Some emulators
+ * and flash carts support writing to RTC but will not persist the date/time
+ * after resetting or powering-off.
```

Comment:
This paragraph is a little generic. Given that there's a handful of supported cartridges / emulators that are relevant, I'd simply list them all with their current status (mentioning version number, or a general "as of July 2021")

Reply:
I'd drop the very first sentence. Don't create excuses for emulator writers 😃

Reply:
I included the first sentence to set expectations for homebrew developers that depending on RTC functionality will inevitably limit where their software will correctly run (at the moment). RTC is currently an esoteric novelty that really needs more momentum behind it before emulators are incentivized to fully support it.

Reply:
ok